docs/proposals/20191030-machine-health-checking.md

```diff
@@ -105,14 +104,17 @@ As an operator of a Management Cluster, I want my machines to be self-healing an
 
 ### Implementation Details/Notes/Constraints
 
-#### Machine annotation:
+#### Machine conditions:
 ```go
-const MachineUnhealthyAnnotation = "machine.cluster.x-k8s.io/unhealthy"
+const ConditionHealthCheckSucceeded ConditionType = "HealthCheckSucceeded"
+const ConditionOwnerRemediationSucceeded ConditionType = "OwnerRemediationSucceeded"
 ```
 
-This annotation is applied by the MHC controller and it is expected that it
-will never be removed. The only current mediation strategy is deletion, which
-is to be handled by the machine's owning controller.
+Both of these conditions are applied by the MHC controller. ConditionHealthCheckSucceeded should only be updated by the MHC after running the health check.
```

Comment:
I take it this is related to [your comment](https://github.com/kubernetes-sigs/cluster-api/pull/3017#discussion_r426704167) about relaxing the need for truthy conditions always being "good"? So you mean `ConditionHealthCheckFailed`?

Reply:
Indeed, I'm just thinking of the fact that we aren't really able to verify that a Machine is actually Healthy, we are looking to trigger behavior on the case that the health check fails rather than succeeds, and we aren't really trying to block any additional behavior/processes until the health check passes.